### PR TITLE
Re-implement validation

### PR DESCRIPTION
**Pull request by matthias-ronge**
The new validation uses the new ruleset for validation.
Please note the explanations in the individual commits. I have tried to better explain why I made the respective changes.

This pull request is based on #2145

### CODE REVIEW

**Comment by Kathrin-Huber** [comment]
please rebase for better review

File: Kitodo-DataEditor/src/main/java/org/kitodo/dataeditor/ruleset/AbstractKeyView.java [hunk 1]
```diff
@@ -31,6 +31,11 @@
      */
     protected final U universal;
 
+    /**
+     * The universal rule.
```
Comment (Kathrin-Huber) [inline Kitodo-DataEditor/src/main/java/org/kitodo/dataeditor/ruleset/AbstractKeyView.java]: For what?

File: Kitodo/src/main/java/org/kitodo/production/helper/batch/BatchTaskHelper.java [hunk 2]
```diff
@@ -455,7 +455,6 @@ private boolean isTaskValid(Task task) throws IOException {
     private boolean invalidMetadataExists(Task task) {
         if (task.isTypeMetadata()
                 && ConfigCore.getBooleanParameterOrDefaultValue(ParameterCore.USE_META_DATA_VALIDATION)) {
-            ServiceManager.getMetadataValidationService().setAutoSave(true);
```
Comment (Kathrin-Huber) [inline Kitodo/src/main/java/org/kitodo/production/helper/batch/BatchTaskHelper.java]: why do you remove it? Has it something to do with validation?

File: Kitodo/src/main/resources/messages/errors_en.properties [hunk 3]
```diff
@@ -86,18 +86,18 @@ metadataDigitalDocumentError=DigitalDocument error at\: {0}
...
+metadataInvalidData=The input "{0}" in field {1} is invalid.
 metadataIsEmpty=contains no data
-metadataMandatoryElement=missing input for\:
-metadataMissingIdentifier=missing value for identifier
+metadataMandatoryElement=Missing input for {0}.
+metadataMediaError=A structure has media assigned which aren't declared.
+metadataMediaUnassigned=The media unit is not assigned to a structure\:
+metadataMissingIdentifier=Missing value for the identifier.
 metadataNotAllowed={0} {1} is not allowed in new element {1}.
-metadataNotEnoughElements=have to exist at least once
```
Comment (Kathrin-Huber) [inline Kitodo/src/main/resources/messages/errors_en.properties]: The following error messages seems to be incomplete!?

File: Kitodo-DataEditor/src/main/java/org/kitodo/dataeditor/ruleset/AbstractKeyView.java (hunk 1 above)
Comment (matthias-ronge) [reply Kitodo-DataEditor/src/main/java/org/kitodo/dataeditor/ruleset/AbstractKeyView.java]: To get the `minOccurs` and `maxOccurs` value.

File: Kitodo/src/main/java/org/kitodo/production/helper/batch/BatchTaskHelper.java (hunk 2 above)
Comment (matthias-ronge) [reply Kitodo/src/main/java/org/kitodo/production/helper/batch/BatchTaskHelper.java]: Here, a variable is set, which has no further meaning. It is not used anywhere.

File: Kitodo/src/main/resources/messages/errors_en.properties (hunk 3 above)
Comment (matthias-ronge) [reply Kitodo/src/main/resources/messages/errors_en.properties]: Exactly. That's why I changed it.

**Comment by matthias-ronge** [comment]
Is that possible? Then the validation module would have to load the ruleset module. As far as I know, that does not work, right?

**Comment by Beacze** [comment]
Since when it doesn't work? I have used command module inside file management module in May 2018. I think since then until now it works.

File: Kitodo/src/main/java/org/kitodo/production/helper/batch/BatchTaskHelper.java (hunk 2 above)
Comment (Kathrin-Huber) [reply Kitodo/src/main/java/org/kitodo/production/helper/batch/BatchTaskHelper.java]: I'm just wondering, why you changed it in this PR

**Comment by Kathrin-Huber** [comment]
You can of course load modules in other modules :)

**Comment by matthias-ronge** [comment]
Well, I did not know that. I always thought the modules were independent. Of course that makes things very different.

**Comment by matthias-ronge** [comment]
I need a lot of methods that are only in the core. Do I have to duplicate all of them into the module?

- `metsService.createMetsXmlElementAccess()`, `fileService.tryLock()`, `fileService.read()` − to read METS file
- `rulesetManagementService.getRulesetManagement()`, `userService.getAuthenticatedUser().getMetadataLanguage()`, `Helper.getRequestParameter()` − to access the rule set
- `ConfigCore.getParameterOrDefaultValue()`, `ParameterCore.VALIDATE_IDENTIFIER_REGEX` − to validate the identifier against configured regex
- `processService.findAll()` − to check the identifier is not used in other process
- `Helper.getTranslation()` − to translate validation messages

**Comment by matthias-ronge** [comment]
Here is a commit which shows the problem. Also I need `ProcessDTO` from the core to tell the title of the conflicting process in an error message.

**Comment by Kathrin-Huber** [comment]
please resolve conflicts

File: Kitodo/src/main/java/org/kitodo/production/services/validation/MetadataValidationService.java [hunk 4]
```diff
@@ -12,32 +12,120 @@
...
+
+    /**
+     * Message key if there are too little entries of a type.
+     */
+    private static final String MESSAGE_VALUE_TOO_RARE = "metadataNotEnoughElements";
+
+    private final MetadataValidationInterface metadataValidation;
+
+    public MetadataValidationService() {
+        metadataValidation = getValidationModule();
```
Comment (Kathrin-Huber) [inline Kitodo/src/main/java/org/kitodo/production/services/validation/MetadataValidationService.java]: Please name it metadatavalidationmodule

**Pull request merged**